src/juju/index.js

```diff
@@ -15,18 +17,28 @@ const options = {
 
 const controllerURL = process.env.REACT_APP_CONTROLLER_URL;
 
+/**
+  Connects to the controller at the url defined in the REACT_APP_CONTROLLER_URL
+  environment variable.
+  @returns {Object} conn The controller connection instance.
+*/
```

Comment:
As per team practices on [commenting code](https://canonical-web-and-design.github.io/practices/coding/javascript.html#commenting-code), we should not use JSdocs formatting unless for library code. I know you're probably not going to be happy about this but using it here and not elsewhere will leave the codebase inconsistent.

Reply:
I wholly disagree with that statement. But if you'd like, this is effectively a library that the main application uses to communicate with jujulib via redux mechanisms 😄

Reply:
> As per team practices on commenting code, we should not use JSdocs formatting unless for library code

That's not what practices says - it says JSDoc should be used for library code (an important requirement for generating API docs), but doesn't stipulate whether it can or should be used elsewhere. We use JSdoc all over the place in maas-ui and crbs-ui for any sufficiently complex function. I don't see how *more* documentation could ever be a bad thing.

Reply:
@squidsoup That discussion came about as it was pointed out that 'JSdoc all over the place ' bear no relation to the code that follows them as they were not being kept updated.

I don't disagree with "more documentation is always good" but imo using JSDocs format anywhere gives the impression that comprehensive documentation could be generated when that is not the case.